packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.test.tsx

```diff
@@ -0,0 +1,434 @@
+    // expect the last call to contain null
+    expect(mockChangeHandler.mock.calls[1][0].detail).toEqual({
+      selectedItem: null,
+    });
+
+    expectMenuToBeClosed();
+    autocompleteRoot.removeEventListener('ds-change', mockChangeHandler);
+  });
+
+  it('should select list items by keyboard', () => {
+    renderAutocomplete();
+
+    const autocompleteRoot = document.querySelector('ds-autocomplete');
+    const mockChangeHandler = jest.fn();
+    autocompleteRoot.addEventListener('ds-change', mockChangeHandler);
+
+    const autocompleteField = screen.getByRole('combobox') as HTMLInputElement;
+    userEvent.click(autocompleteField);
+    userEvent.type(autocompleteField, 'c');
+    userEvent.type(autocompleteField, '{arrowdown}');
+    userEvent.type(autocompleteField, '{enter}');
+
+    expect(autocompleteField.value).toBe('Cook County, IL');
+    expect(mockChangeHandler).toHaveBeenCalledTimes(1);
+    expect(mockChangeHandler.mock.calls[0][0].detail).toEqual({
+      selectedItem: JSON.parse(defaultItems)[0],
+    });
+
+    expectMenuToBeClosed();
+    autocompleteRoot.removeEventListener('ds-change', mockChangeHandler);
+  });
+
+  it('should not call onChange when an item was not selected', () => {
+    renderAutocomplete();
+
+    const autocompleteRoot = document.querySelector('ds-autocomplete');
+    const mockChangeHandler = jest.fn();
+    autocompleteRoot.addEventListener('ds-change', mockChangeHandler);
+
+    const autocompleteField = screen.getByRole('combobox') as HTMLInputElement;
+    userEvent.click(autocompleteField);
+    userEvent.type(autocompleteField, 'c');
+    userEvent.type(autocompleteField, 'o');
+    userEvent.type(autocompleteField, '{enter}');
+    expect(mockChangeHandler).not.toHaveBeenCalled();
+
+    autocompleteRoot.removeEventListener('ds-change', mockChangeHandler);
+  });
+
+  it('should clear the input value by keyboard', () => {
+    renderAutocomplete();
+    const autocompleteField = screen.getByRole('combobox') as HTMLInputElement;
+    autocompleteField.focus();
+    userEvent.click(autocompleteField);
+    userEvent.type(autocompleteField, 'c');
+    userEvent.type(autocompleteField, '{arrowdown}');
+    userEvent.type(autocompleteField, '{enter}');
+
+    expect(autocompleteField.value).toBe('Cook County, IL');
+
+    userEvent.tab();
+
+    const clearButton = screen.getByText('Clear search');
+    userEvent.click(clearButton);
+
+    expect(autocompleteField.value).toBe('');
+  });
+
+  it('closes the listbox when ESC is pressed', () => {
+    renderAutocomplete();
+    const autocompleteField = screen.getByRole('combobox') as HTMLInputElement;
+    userEvent.click(autocompleteField);
+    userEvent.type(autocompleteField, 'c');
+
+    expectMenuToBeOpen();
+
+    expect(autocompleteField.value).toEqual('c');
+
+    userEvent.type(autocompleteField, '{esc}');
+
+    expectMenuToBeClosed();
+  });
+
+  // it("calls child TextField's event handlers", () => {
```

Comment:
Several unit tests are commented out. Is this intentional, or should we consider re-enabling or removing them?

Reply:
The unit tests that are commented out correspond to how the react component's API expects a `<TextField />` component to be passed in as a child. The new, simpler web component API does not currently allow for that functionality. Instead, we're "flattening" the component and passing in a far less customizable text input here: https://github.com/CMSgov/design-system/blob/7735a395f977b258669ab8c6574f16e7669e5402/packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.tsx#L66-L77

I may spin up a follow-up ticket to see if there's a need to adjust the API to "unflatten" it and have it match the React component's API more closely.

Reply:
https://jira.cms.gov/browse/WNMGDS-2992